vyperlang/vyper
# chore: remove gas estimates from ABI

### What I did

Fix for #2151.

### How I did it

Remove the step to add gas estimates to the ABI in `build_abi_output` function in `vyper/compiler/output.py`.

### How to verify it

See test cases.

Since gas estimates are removed from the ABI, it no longer makes sense to calculate the `gas` value for function calls in test cases based on the ABI gas estimate + 50000. Instead, we default to the standard gas estimates. This however results in certain test cases for `assert` statements failing in `tests/parser/features/test_assert_unreachable.py` and `tests/parser/features/test_assert.py` (the error is due to an error in parsing the exception in `eth_tester`). The workaround is to supply a `gas` value to the `call` kwarg (still not sure why supplying this parameter changes the response returned).

### Description for the changelog

Remove gas estimates from ABI output

### Cute Animal Picture

![Put a link to a cute animal picture inside the parenthesis-->](https://npr.brightspotcdn.com/dims4/default/e1783a8/2147483647/strip/true/crop/5312x2988+0+0/resize/880x495!/quality/90/?url=http%3A%2F%2Fnpr-brightspot.s3.amazonaws.com%2Flegacy%2Fsites%2Fwjct%2Ffiles%2F201812%2Fbaby_otter.jpg)
